pkg/kubelet/eviction/helpers.go

```diff
@@ -847,10 +849,30 @@ func makeSignalObservations(summary *statsapi.Summary) (signalObservations, stat
 	result := signalObservations{}
 
 	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil && memory.WorkingSetBytes != nil {
-		result[evictionapi.SignalMemoryAvailable] = signalObservation{
-			available: resource.NewQuantity(int64(*memory.AvailableBytes), resource.BinarySI),
-			capacity:  resource.NewQuantity(int64(*memory.AvailableBytes+*memory.WorkingSetBytes), resource.BinarySI),
-			time:      memory.Time,
+		if runtime.GOOS == "windows" {
+			// Don't use stats summary to compute available memory on Windows because it does not take into account
+			// system process memory usage (which can be significant). Instead use win32 api's to query global
+			// memory usage.
```

Comment:
Is this an issue with the Stats summary and related to https://github.com/kubernetes/kubernetes/issues/112111?  Should we be updating the stats summary to use the correct value here instead of querying it directly?

Reply:
It has been a few weeks since I last looked at this but I don't think issue was related to Stats summary being incorrect - it just works different from linux. On Windows OS overhead is a lot compared to Linux and it made more sense to just use the perf counters. I'll try and remember exactly why...

Reply:
I am wondering what the node stats are used for in Windows, I thought this would be one of the main use cases.  The reason for asking is I think it would clean up this nested if statements here as well as the check in the tests for being with in a range.

Reply:
I'm going to look into this again

Reply:
I finally got a chance to take a deep look at this and I'm still thinking that using total and available physical memory is the right thing to do for eviction thresholds.

Eviction thresholds can be specified as a percentage of total memory (< 5%) or a value (< 500Mb). Windows always uses virtual memory and trying to determine these values based on what we have in the stats summary / available in perf counters is too unpredictable.

I agree that there can be issues in how memory usage is reported in /stats/summary but don't see an easy way to fix that that would also allow eviction to work predictably.

I wrote a small powershell script to dump memory info (both total / available physical memory and variety of memory / process memory perf counters)

https://gist.githubusercontent.com/marosset/8334544e261d57e756f13514a3202a62/raw/e1b0b5161c18514b3bb6b335de1a2660effef901/memory.ps1

When running on different machines with different amounts of memory here are some example outputs

```console
 .\memory.ps1
Stats from systeminfo / GlobalMemoryStatusEx
  Total Physical Memory Mb:  4095.55078125
  Free Physical Memory Mb :  1903.83984375
  Percent Usage           :  53.5144368746191

Status from perf counters
Available Bytes MB               : 1885.375
Committed Bytes MB               : 2426.89453125
Cache Bytes MB                   : 87.328125
Working Set (total) MB           : 2374.515625
Working Set - Private (total) MB : 992.08203125
Private (total) MB               : 1672.94140625

Available Bytes + committed Bytes MB         :  4312.26953125
Available Bytes + Working Set MB             :  4259.890625
Available Bytes + Working Set - Cached MB    :  4172.5625
Available Bytes + Private Bytes MB           :  3558.31640625
Available Bytes + (Working Set - Private) MB :  2877.45703125
```
 
and 

```console
.\memory.ps1
Stats from systeminfo / GlobalMemoryStatusEx
  Total Physical Memory Mb:  65401.28125
  Free Physical Memory Mb :  24365.81640625
  Percent Usage           :  62.744129869397625

Status from perf counters
Available Bytes MB               : 24381.48046875
Committed Bytes MB               : 55918.75390625
Cache Bytes MB                   : 920.1328125
Working Set (total) MB           : 44125.765625
Working Set - Private (total) MB : 31531.4765625
Private (total) MB               : 48290.07421875

Available Bytes + committed Bytes MB         :  80300.234375
Available Bytes + Working Set MB             :  68507.24609375
Available Bytes + Working Set - Cached MB    :  67587.11328125
Available Bytes + Private Bytes MB           :  72671.5546875
Available Bytes + (Working Set - Private) MB :  55912.95703125
```

I couldn't find a combination of metrics reported from perf metrics that reasonably estimate memory usage as close as what this PR currently does.

@jsturtevant - let's discuss in the next SIG-Windows meeting if you'll be able to attend.